### PR TITLE
improvement: add do not show bsp errors option

### PR DESCRIPTION
**Pull request by kasiaMarek**
Add possibility to stop showing bsp errors.

### CODE REVIEW

File: metals/src/main/scala/scala/meta/internal/builds/BSPErrorHandler.scala [hunk 1]
```diff
@@ -62,6 +67,8 @@ class BspErrorHandler(
         restartBspServer().ignoreValue
       case BspErrorHandler.dismiss =>
         Future.successful(dismissedErrors.add(message)).ignoreValue
+      case BspErrorHandler.doNotShowErrors =>
+        Future.successful { doNotShowErrors = true }
```
[the remark this answers is not in the document]
Comment (kasiaMarek) [reply metals/src/main/scala/scala/meta/internal/builds/BSPErrorHandler.scala:71]: Don't we want to move this long term to bsp status + doctor anyway? Then this flag won't be needed anyway.

[the remark this answers is not in the document]
Comment (tgodzik) [reply metals/src/main/scala/scala/meta/internal/builds/BSPErrorHandler.scala:71]: Hmm.... in that case yeah. Are we able to sensibly move it to status? We could add the option to disable for now until this is implemented

File: metals/src/main/resources/db/migration/V5__Show_bsp_errors.sql [hunk 2]
```diff
@@ -0,0 +1,4 @@
+-- If should show bsp errors
+create table show_bsp_errors(
```
Comment (tgodzik) [inline metals/src/main/resources/db/migration/V5__Show_bsp_errors.sql:2]: We could add it to the notifications table, no?

Comment (kasiaMarek) [reply metals/src/main/resources/db/migration/V5__Show_bsp_errors.sql]: Somehow I wasn't aware we had such a generic mechanism for notifications.

**Review by tgodzik — APPROVED** [review]
LGTM!

**Pull request merged**